scripts/devops_tasks/verify_autorest.py

```diff
@@ -54,7 +55,31 @@ def find_swagger_folders(directory):
     return ret
 
 
-def check_diff(folder):
+def check_for_open_pr(service_directory):
+    # Return True if there is an OPEN PR of the format ""Automated autorest generation for {service_directory}""
+    # Return False otherwise
+
+    pr_title = "Automated autorest generation for {}".format(service_directory)
+
+    page_size = 50
+    page_number = 1
+    request_url = "https://api.github.com/repos/Azure/azure-sdk-for-python/pulls?state=open&per_page={}&page={}"
+
+    s = requests.session()
+    response = s.get(request_url.format(page_size, page_number))
+    while "next" in response.links:
+        body = response.content.decode("utf-8")
+        body = json.loads(body)
+        for link in body:
+            if link["title"] == pr_title:
+                return True
+
+        next_page = response.links["next"]["url"]
+        response = s.get(next_page)
+    return False
+
+
+def check_diff(folder, service_dir):
```

Comment:
Similar you shouldn't need to manually do a git diff.  See https://github.com/Azure/azure-sdk-for-python/blob/master/eng/common/pipelines/templates/steps/git-push-changes.yml#L18

Reply:
The git diff is for users to be able to inspect which files are different in the pipeline on a failure.

Reply:
OK if that is interesting for other cases that is fine. However I'd encourage you to do those type of command line calls inline script blocks in the yml file, unless you feel folks would want to run them locally as well. 

This is perhaps my preference for using a shell scripting language (script/bash/pwsh) when running command line tools as opposed to wrapping them in process calls in a higher level language like python. Just something to keep in mind not something that is necessarily actionable. :)